docs/specifications/tests/DNSSEC-TP/dnssec15.md

```diff
@@ -0,0 +1,176 @@
+CDNSKEY.
+
+If a name server has issues covered by [Basic04] (basic name server
+issues) no messages will be outputted from this test case.
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Summary
+
+* [INFO] message if CDS RRset is found on any name server for the zone.
+* [INFO] message if CDNSKEY RRset is found on any name server for the zone.
+* [NOTICE] message if only one type is found.
+* [ERROR] message if the two RRsets (CDS and CDNSKEY) both exist but do not 
+  match.
+* [ERROR] message if not all servers have the same CDS RRset (if CDS exists)
+* [ERROR] message if not all servers have the same CDNSKEY RRset (if CDNSKEY
+  exists)
+
+## Ordered description of steps to be taken to execute the test case
+
+1.  Create a CDS query with EDNS disabled for the apex of the
+    *Child Zone*.
+
+2.  Create a CDNSKEY query with EDNS disabled for the apex of the
+    *Child Zone*.
+
+4.  Retrieve all name server IP addresses for the *Child Zone* using
+    [Get-Del-NS-IPs] and [Get-Zone-NS-IPs] ("NS IP").
+
+5.  Create the following empty sets:
+    1. Name server IP address and associated CDS RRset ("CDS RRsets"). A
+       name server IP can hold an empty RRset.
+    2. Name server IP address and associated CDNSKEY RRset ("CDNSKEY RRsets"). 
+       A name server IP can hold an empty RRset.
+    3. Name server IP set ("Mismatch CDS/CDNSKEY").
+
+6.  Repeat the following steps for each name server IP address in *NS IP*:
+
+    1. Send the CDS query over UDP to the name server IP address.
+       1. If no DNS response is returned, then go to next name server IP.
+       2. Else, if AA bit is not set in the DNS response, then go to next 
+          name server IP.
+       3. Else, if the RCODE in the DNS response is not *NOERROR*, then go to
+          next name server IP.
+       4. Else, if the DNS response contains no CDS record in the
+          answer section, then add the name server IP and an empty RRset to
+          *CDS RRsets*.
+       5. Else, add the name server IP and the CDS RRset from the answer
+          section to *CDS RRsets*.
+    2. Send the CDNSKEY query over UDP to the name server IP address.
+       1. If no DNS response is returned, then go to next name server IP.
+       2. Else, if AA bit is not set in the DNS response, then go to next
+          name server IP.
+       3. Else, if the RCODE in the DNS response is not *NOERROR*, then go to
+          next name server IP.
+       4. Else, if the DNS response contains no CDNSKEY record in the
+          answer section, then add the name server IP and an empty RRset to
+          *CDNSKEY RRsets* and go to next name server IP.
+       5. Else, add the name server IP and the CDNSKEY RRset from the answer
+          section to *CDNSKEY RRsets* and go to next name server IP.
+
+7.  If neither *CDS RRsets* nor *CDNSKEY RRsets* has any non-empty RRset
+    then output *[DS15_NO_CDS_CDNSKEY]* and terminate this test case.
+
```

Comment:
Instead of the negative definition, do a positive one, easier to understand.

If all *CDS RRsets* and all *CDNSKEY RRsets* are empty thenoutput *[DS15_NO_CDS_CDNSKEY]* and terminate test case.

Reply:
Resolved.